python-package/lightgbm/basic.py

```diff
@@ -135,7 +135,12 @@ def param_dict_to_str(data):
     pairs = []
     for key, val in data.items():
         if isinstance(val, (list, tuple, set)) or is_numpy_1d_array(val):
-            pairs.append(str(key) + '=' + ','.join(map(str, val)))
+            def to_string(x):
+                if isinstance(x, list):
+                    return '[' + ','.join(map(str, x)) + ']'
```

Comment:
```suggestion
                    return "[{}]".format(','.join(map(str, x)))
```

I think this would be a little easier to read using string interpolation. I recommend  `.format()` since it's an option supported in Python 2 and 3. @StrikerRUS what do you think?